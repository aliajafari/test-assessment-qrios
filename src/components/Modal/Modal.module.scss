@keyframes backgroundAnimation {
    from {
        background-color: rgba(0,0,0,0);
    }

    to {
        background-color: rgba(0,0,0,0.8);
        animation-play-state: paused;
    }
}

@keyframes containerAnimation {
    from {
        opacity: 0;

    }

    to {
        opacity: 1;
    }
}

.modalWrapper {
    position: fixed;
    z-index: 10;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    // background-color: rgba(0, 0, 0, 0.8);
    overflow-x: scroll;
    animation-name: backgroundAnimation;
    animation-duration: 0.3s;
    animation-fill-mode: forwards;
}

.modalContainer {
    opacity: 0;
    margin: 0;
    position: absolute;
    top: 50%;
    left: 50%;
    -ms-transform: translate(-50%, -50%);
    transform: translate(-50%, -50%);
    max-width: 600px;
    background-color: black;
    padding: 15px;
    color: white;
    animation-name: containerAnimation;
    animation-duration: 0.4s;
    animation-delay: 0.4s;
    animation-fill-mode: forwards;
    >div {
        max-height: 400px;

        img {
            display: block;
            height: 100%;
            margin: auto;
        }
    }

    h3 {
        font-size: 16px;
    }

    p {
        margin-top: 0;
    }
}