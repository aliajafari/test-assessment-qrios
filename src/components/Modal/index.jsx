import React from "react";
import styles from "./Modal.module.scss";

function Modal({ title, image, description, onClickClose }) {
  const onClickBackground = (e) => {
    if (e.target.className === styles.modalWrapper) {
      onClickClose();
    }
  };

  return (
    <div className={styles.modalWrapper} onClick={onClickBackground} data-testid="close-modal">
      <div className={styles.modalContainer} data-testid="modal-container">
        <div>
          <img data-testid="modal-image" src={image} alt="" />
        </div>
        <h3 data-testid="modal-title">{title}</h3>
        <p data-testid="modal-description">{description}</p>
      </div>
    </div>
  );
}

export default Modal;
