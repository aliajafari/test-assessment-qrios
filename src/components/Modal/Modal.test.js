import React from 'react';
import renderer from 'react-test-renderer';
import Modal from './index';
import ReactDOM from 'react-dom'
import { render, fireEvent, screen } from '@testing-library/react'

const modalRoot = document.createElement('div')
modalRoot.setAttribute('id', 'portal-root')
document.body.appendChild(modalRoot)

const data = {
    "id": 11,
    "title": "Lorem ipsum dolor sit amet, consectetur adipiscing elit",
    "description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.",
    "image": "https://via.placeholder.com/300x300/09f/fff.png"
}

describe('Modal =>', () => {
    test('renders correctly when there is a Modal', () => {
        const tree = renderer.create(<Modal {...data} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('modal shows and a close button', () => {
        const handleClose = jest.fn()
        render(<Modal {...data} onClickClose={handleClose} />)
        expect(screen.getByTestId('modal-container')).toBeTruthy()
        fireEvent.click(screen.getByTestId('close-modal'))
        expect(handleClose).toHaveBeenCalledTimes(1)
    })


    test('Modal shows with data', () => {
        const handleClose = jest.fn()
        render(<Modal {...data} onClickClose={handleClose} />)
        expect(screen.getByTestId('modal-title')).toHaveTextContent(data.title)
        expect(screen.getByTestId('modal-image')).toHaveAttribute('src', data.image)
        expect(screen.getByTestId('modal-description')).toHaveTextContent(data.description)
    })
})