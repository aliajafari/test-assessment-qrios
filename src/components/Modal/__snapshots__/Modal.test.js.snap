// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal => renders correctly when there is a Modal 1`] = `
<div
  className="modalWrapper"
  data-testid="close-modal"
  onClick={[Function]}
>
  <div
    className="modalContainer"
    data-testid="modal-container"
  >
    <div>
      <img
        alt=""
        data-testid="modal-image"
        src="https://via.placeholder.com/300x300/09f/fff.png"
      />
    </div>
    <h3
      data-testid="modal-title"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit
    </h3>
    <p
      data-testid="modal-description"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
    </p>
  </div>
</div>
`;
